src/session.rs

```diff
@@ -12,31 +17,38 @@ use uuid::Uuid;
 ///
 /// This is Auto generated by the Session Layer Upon Service Execution.
 #[derive(Debug, Clone)]
-pub struct AxumSession {
-    pub(crate) store: AxumSessionStore,
+pub struct AxumSession<T>
+where
+    T: AxumDatabasePool + Clone + fmt::Debug + std::marker::Sync + std::marker::Send + 'static,
+{
+    pub(crate) store: AxumSessionStore<T>,
     pub(crate) id: AxumSessionID,
 }
 
 /// Adds FromRequest<B> for AxumSession
 ///
 /// Returns the AxumSession from Axums request extensions.
 #[async_trait]
-impl<B> FromRequest<B> for AxumSession
+impl<B, T> FromRequest<B> for AxumSession<T>
 where
     B: Send,
+    T: AxumDatabasePool + Clone + fmt::Debug + std::marker::Sync + std::marker::Send + 'static,
```

Comment:
we really should try to simplfy the Sync, Send and Debug to clean up the code a bit in all the where statements. we can do this by consalidating a use statement like std::{fmt::debug, marker::{Sync, Send}}